src/module.c

```diff
@@ -6664,12 +6657,8 @@ RedisModuleCtx *RM_GetThreadSafeContext(RedisModuleBlockedClient *bc) {
  * a long term, for purposes such as logging. */
 RedisModuleCtx *RM_GetDetachedThreadSafeContext(RedisModuleCtx *ctx) {
     RedisModuleCtx *new_ctx = zmalloc(sizeof(*new_ctx));
-    RedisModuleCtx empty = REDISMODULE_CTX_INIT;
-    memcpy(new_ctx,&empty,sizeof(empty));
-    new_ctx->module = ctx->module;
-    new_ctx->flags |= (REDISMODULE_CTX_THREAD_SAFE |
-                       REDISMODULE_CTX_USES_TEMP_CLIENT);
-    new_ctx->client = moduleAllocTempClient();
+    moduleCreateContext(new_ctx, ctx->module,
+                        REDISMODULE_CTX_THREAD_SAFE|REDISMODULE_CTX_NEW_CLIENT);
```

Comment:
maybe refer to the comment we're adding in RM_GetThreadSafeContext

Reply:
added a comment